src/NuGet.Clients/NuGet.PackageManagement.PowerShellCmdlets/NuGetPowerShellBaseCommand.cs

```diff
@@ -1095,7 +1099,21 @@ public void ReportError(string message)
 
         public NuGetActionType ActionType { get; set; }
 
-        public INuGetTelemetryService TelemetryService { get; set; }
+        public KeyValuePair<string, Guid> OperationId
```

Comment:
dont you just need a simple Guid OperationId which is generated new for each operaiton? why would you need setter here?

Reply:
this is for UI, all UI operations share same UI project context, so we need to set the id later